### PR TITLE
feat: fallback metadata injection

### PR DESCRIPTION
**Pull request by LunrEclipse**
## Description

Created a function in root directory to take any page that doesn't have custom metadata and injects a default with a title and image. 

_Linear ticket:_ WEB-2604

### Automated testing

- [x] Unit test - Tested with `yarn cloud:tests` and added a test suite `default.test.ts`

### CODE REVIEW

File: functions/collection.test.ts [hunk 1]
```diff
@@ -38,9 +38,7 @@ test.each(collections)('should inject metadata for valid collections', async (co
 
 const invalidCollections = [
   'http://127.0.0.1:3000/nfts/collection/0xed5af388653567af2f388e6224dc7c4b3241c545',
-  'http://127.0.0.1:3000/nfts/collection/0xed5af388653567af2f388e6224dc7c4b3241c545/10',
```
Comment (zzmp) [inline functions/collection.test.ts:41]: Why are you removing these? It seems unrelated.

File: functions/tsconfig.json [hunk 2]
```diff
@@ -15,5 +15,5 @@
     "baseUrl": "functions"
   },
   "exclude": ["../node_modules", "../src"],
-  "include": ["**/*.ts"],
+  "include": ["**/*.ts", ".ts"],
```
Comment (zzmp) [inline functions/tsconfig.json:18]: What does this change do?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: functions/collection.test.ts (hunk 1 above)
Comment (LunrEclipse) [reply functions/collection.test.ts:41]: those tests actually get caught by the fallback so it caused them to start failing

File: functions/[[index]].ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const data = {
+    title: 'Uniswap Interface',
+    image: imageUri,
+    url: request.url,
+  }
+  try {
+    return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(await next())
+  } catch (e) {
+    return next()
+  }
```
Comment (zzmp) [inline functions/[[index]].ts:15]:
```suggestion
  const res = next()
  try {
    return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(await res)
  } catch (e) {
    return res
  }
```

File: functions/[[index]].ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { MetaTagInjector } from './components/metaTagInjector'
+
+// eslint-disable-next-line import/no-unused-modules
```
Comment (zzmp) [inline functions/[[index]].ts:3]: Do you need this comment?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: functions/[[index]].ts (hunk 4 above)
Comment (LunrEclipse) [reply functions/[[index]].ts:3]: yeye its just like the other functions

**Review by natew — APPROVED** [review]
functions are cool

**Review by zzmp — APPROVED** [review]
